applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_plane_stress_uncoupled_shear.cpp

```diff
@@ -0,0 +1,108 @@
+
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+ElasticIsotropicPlaneStressUncoupledShear::ElasticIsotropicPlaneStressUncoupledShear()
+    : LinearPlaneStress()
+{
+}
+
+//******************************COPY CONSTRUCTOR**************************************
+//************************************************************************************
+
+ElasticIsotropicPlaneStressUncoupledShear::ElasticIsotropicPlaneStressUncoupledShear(const ElasticIsotropicPlaneStressUncoupledShear& rOther)
+    : LinearPlaneStress(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer ElasticIsotropicPlaneStressUncoupledShear::Clone() const
+{
+    ElasticIsotropicPlaneStressUncoupledShear::Pointer p_clone(new ElasticIsotropicPlaneStressUncoupledShear(*this));
+    return p_clone;
+}
+
+//*******************************DESTRUCTOR*******************************************
+//************************************************************************************
+
+ElasticIsotropicPlaneStressUncoupledShear::~ElasticIsotropicPlaneStressUncoupledShear()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+int ElasticIsotropicPlaneStressUncoupledShear::Check(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+)
+{
+    ElasticIsotropic3D::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+
+    if (SHEAR_MODULUS.Key() == 0 || rMaterialProperties[SHEAR_MODULUS] <= 0.0)
+    {
+        KRATOS_ERROR << "SHEAR_MODULUS has Key zero or invalid value " << std::endl;
+    }
+
+    return 0;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void ElasticIsotropicPlaneStressUncoupledShear::CalculateElasticMatrix(Matrix& C, Parameters& rValues)
+{
+    const Properties& MaterialProperties = rValues.GetMaterialProperties();
+    const double& E = MaterialProperties[YOUNG_MODULUS];
+    const double& NU = MaterialProperties[POISSON_RATIO];
+    const double& G = MaterialProperties[SHEAR_MODULUS];
+    const double& G1 = MaterialProperties[SHEAR_MODULUS_GAMMA12];
+    const double& G2 = MaterialProperties[SHEAR_MODULUS_GAMMA12_2];
+    const double& G3 = MaterialProperties[SHEAR_MODULUS_GAMMA12_3];
+    const double& G4 = MaterialProperties[SHEAR_MODULUS_GAMMA12_4];
+
+    const Vector& StrainVector = rValues.GetStrainVector();
```

Comment:
r_strain_vector => please have a look at the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide#coding-convention)

Reply:
I have to put r_e; r_nu; r_g or we keep some exceptions?

Reply:
For these ones it is ok from my side to keep them as they are, since they are very short.